lms/services/grading_info.py

```diff
@@ -43,27 +43,33 @@ def get_by_assignment(self, oauth_consumer_key, context_id, resource_link_id):
         :param resource_link_id: the assignment's resource_link_id
             (identifies the assignment within the LMS course)
         :type resource_link_id: str
-
-        :return: a list of all the (GradingInfo, HUser) tuples for the assignment
-        :rtype: list[(GradingInfo, HUser)]
         """
         grading_infos = self._db.query(GradingInfo).filter_by(
             oauth_consumer_key=oauth_consumer_key,
             context_id=context_id,
             resource_link_id=resource_link_id,
         )
 
-        return [
-            (
-                grading_info,
-                HUser(
-                    authority=self._authority,
-                    username=grading_info.h_username,
-                    display_name=grading_info.h_display_name,
-                ),
+        # The list of "student" dicts that we'll return.
+        students = []
+
+        # Create a "student" dict for each GradingInfo.
+        for grading_info in grading_infos:
+            h_user = HUser(
+                authority=self._authority,
+                username=grading_info.h_username,
+                display_name=grading_info.h_display_name,
             )
-            for grading_info in grading_infos
-        ]
+            students.append(
+                {
+                    "userid": h_user.userid,
+                    "displayName": h_user.display_name,
+                    "LISResultSourcedId": grading_info.lis_result_sourcedid,
+                    "LISOutcomeServiceUrl": grading_info.lis_outcome_service_url,
```

Comment:
Is it worth mentioning that these are the exact config params here? It would explain why the casing is so goofy. My instinct seeing this would be "let's fix this!"